# NE-EpiGuard: Waterborne Disease Outbreak Prediction — Northeast India
## Exploratory Data Analysis (EDA)

---

## 1. Problem Statement

### What is a Waterborne Disease Outbreak?
Waterborne diseases are illnesses caused by pathogenic microorganisms transmitted through contaminated water. In Northeast India, a region characterized by extreme monsoon patterns, high flooding frequency, and significant variation in sanitation infrastructure across its 8 states, waterborne diseases like Cholera, Typhoid, Dysentery, Hepatitis A/E, Leptospirosis, and Giardiasis remain a persistent public health challenge.

### Why Northeast India?
The Northeast region contributes disproportionately to India's waterborne disease burden:
- **Extreme monsoon seasonality** — some districts receive 1000mm+ rainfall in June–August alone
- **High flooding incidence** — especially in Assam, Meghalaya, and Tripura floodplains
- **Sanitation gaps** — open defecation rates and untreated water sources remain elevated in remote districts
- **Geographic remoteness** — many hill districts in Arunachal Pr|adesh, Nagaland, and Mizoram have limited healthcare access, making early prediction critical

### Our Goal
Using patient-level data capturing demographics, water quality parameters, environmental conditions, sanitation indicators, and clinical symptoms across all 8 Northeast Indian states, we aim to:
1. **Understand** the distribution and drivers of each disease class
2. **Identify** which water quality parameters, climate factors, and sanitation indicators most strongly predict disease type
3. **Prepare** a clean, feature-engineered dataset for multi-class disease classification
4. **Power** an AI advisory system that provides health guidance and specialist recommendations

### Disease Classes
| Disease | Transmission Route | Key Risk Factors |
|---|---|---|
| Cholera | Fecal-oral via water | High fecal coliform, flooding, untreated water |
| Typhoid | Contaminated water/food | Poor sanitation, high TDS, open defecation |
| Dysentery | Contaminated water | High turbidity, untreated sources |
| Hepatitis A | Fecal-oral | Poor handwashing, low WQI |
| Hepatitis E | Contaminated water | Flooding, low dissolved oxygen |
| Leptospirosis | Soil/water contact | Flooding, animal contact, monsoon |
| Giardiasis | Contaminated water | Low WQI, surface water sources |
| No Disease | — | Good water quality, sanitation |

---

## 2. Basic Data Checks

In [1]:
# Basic libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# Set background graph style
plt.style.use('dark_background')

# import warnings to ignore the warnings
import warnings
warnings.filterwarnings('ignore')

# Load Dataset
data = pd.read_csv(r'D:\Minor Project\Notebooks\Data\northeast_waterborne.csv')
# 1. Check Shape
print('Data Shape:', data.shape)

# 2. View first few rows
display(data.head())

# 3. Info to see column types and non-null counts
print('\nData Info:')
data.info()

# 4. Summary statistics for numeric columns
print('\nNumeric Summary:')
display(data.describe())

# 5. Unique values per column
print('\nUnique Values Count:')
print(data.nunique())

#6 Checking duplicate rows
print('\n=== DUPLICATE ROWS ===')
print(f'Duplicates: {data.duplicated().sum()}')

Data Shape: (405074, 39)


,state,district,latitude,longitude,population_density,age,gender,water_source,water_treatment,water_quality_index,...,flooding,symptom_diarrhea,symptom_vomiting,symptom_fever,symptom_abdominal_pain,symptom_dehydration,symptom_jaundice,symptom_bloody_stool,symptom_skin_rash,disease
0,Assam,Tinsukia,25.7467,90.2411,505,22,Female,Borewell,Untreated,17.8,...,0,1,0,1,1,1,0,1,0,Dysentery
1,Assam,Darrang,27.0947,90.6179,578,50,Male,Piped,Chlorinated,99.3,...,0,0,0,0,0,0,0,0,0,No_Disease
2,Assam,Cachar,25.1711,93.2949,311,9,Male,Tanker,Untreated,28.3,...,0,0,1,1,0,0,1,0,1,Leptospirosis
3,Manipur,Ukhrul,24.2054,94.1155,179,7,Female,River,Untreated,33.5,...,0,1,1,1,1,1,1,0,0,Hepatitis_A
4,Arunachal Pradesh,Tirap,28.5052,92.9995,49,34,Male,River,Boiled,69.1,...,0,0,0,0,0,0,0,0,0,No_Disease



Data Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 405074 entries, 0 to 405073
Data columns (total 39 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   state                     405074 non-null  object 
 1   district                  405074 non-null  object 
 2   latitude                  405074 non-null  float64
 3   longitude                 405074 non-null  float64
 4   population_density        405074 non-null  int64  
 5   age                       405074 non-null  int64  
 6   gender                    405074 non-null  object 
 7   water_source              405074 non-null  object 
 8   water_treatment           405074 non-null  object 
 9   water_quality_index       405074 non-null  float64
 10  ph                        405074 non-null  float64
 11  turbidity_ntu             405074 non-null  float64
 12  dissolved_oxygen_mg_l     405074 non-null  float64
 13  bod_mg_l                  405074

,latitude,longitude,population_density,age,water_quality_index,ph,turbidity_ntu,dissolved_oxygen_mg_l,bod_mg_l,fecal_coliform_per_100ml,...,avg_humidity_pct,flooding,symptom_diarrhea,symptom_vomiting,symptom_fever,symptom_abdominal_pain,symptom_dehydration,symptom_jaundice,symptom_bloody_stool,symptom_skin_rash
count,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,...,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000,405074.000000
mean,25.708298,92.871637,216.322339,25.758928,41.638291,7.272287,19.874901,6.410288,12.585558,1415.631838,...,73.499393,0.159122,0.447405,0.339617,0.492666,0.540899,0.332853,0.177866,0.145090,0.083925
std,1.679106,1.938430,170.466364,17.940091,25.596080,0.542264,12.907632,3.001760,7.900259,1204.824823,...,14.952101,0.365790,0.497227,0.473580,0.499947,0.498325,0.471235,0.382401,0.352192,0.277276
min,21.500200,88.000000,10.000000,0.000000,10.000000,6.000000,0.000000,2.000000,0.000000,0.000000,...,40.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,24.454800,91.599800,74.000000,12.000000,22.800000,6.860000,7.100000,4.100000,4.500000,90.000000,...,61.300000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,25.660500,92.959000,157.000000,24.000000,31.600000,7.260000,19.800000,5.600000,13.000000,1304.000000,...,73.900000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000
75%,26.925275,94.313900,366.000000,38.000000,64.400000,7.680000,30.400000,7.900000,18.900000,2243.000000,...,86.500000,0.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.000000,0.000000,0.000000
max,29.500000,97.500000,600.000000,90.000000,100.000000,8.500000,50.000000,14.000000,30.000000,5000.000000,...,98.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000



Unique Values Count:
state                           8
district                       84
latitude                    74121
longitude                   81625
population_density            542
age                            91
gender                          2
water_source                    7
water_treatment                 4
water_quality_index           752
ph                            251
turbidity_ntu                 501
dissolved_oxygen_mg_l         121
bod_mg_l                      301
fecal_coliform_per_100ml     4802
total_coliform_per_100ml     9696
tds_mg_l                    19274
nitrate_mg_l                  801
fluoride_mg_l                 301
arsenic_ug_l                  801
open_defecation_rate          952
toilet_access                   2
sewage_treatment_pct          902
handwashing_practice            3
month                          12
season                          4
avg_temperature_c             271
avg_rainfall_mm              8952
avg_humidity_pct          

### Insight
- The dataset contains **405,074 rows and 39 columns** with zero missing values across all columns — no imputation needed.
- Most columns are numeric (float64/int64). `state`, `district`, `gender`, `water_source`, `water_treatment`, `handwashing_practice`, `season`, `disease` are categorical (object).
- **Issue**: `district` has 84 unique values — too high for one-hot encoding, use target encoding or drop in favour of `latitude`/`longitude`.
- **Issue**: `fecal_coliform_per_100ml` and `total_coliform_per_100ml` are likely right-skewed — apply `np.log1p()` transform before scaling.
- **Issue**: `month` is stored as integer (1–12) but is cyclical — apply sine/cosine encoding before modeling.
- All 8 symptom columns are confirmed binary (0/1) — no encoding needed.
- `disease` has 8 unique classes (7 diseases + No_Disease) — confirmed multiclass classification problem.


In [2]:
# Catergorical columns uqique value 

cat_cols = data.select_dtypes(include='object').columns.tolist()

for col in cat_cols:
    print(f'\n{col} ({data[col].nunique()} unique):')
    print(data[col].value_counts())


state (8 unique):
state
Assam                96559
Arunachal Pradesh    62437
Nagaland             53171
Meghalaya            52828
Manipur              43344
Mizoram              38694
Tripura              38582
Sikkim               19459
Name: count, dtype: int64

district (84 unique):
district
Dimapur                  5022
Darrang                  4950
South Sikkim             4930
Lunglei                  4924
North Sikkim             4916
                         ... 
South West Garo Hills    4737
Upper Siang              4734
Imphal West              4724
Thoubal                  4705
Zunheboto                4640
Name: count, Length: 84, dtype: int64

gender (2 unique):
gender
Female    202574
Male      202500
Name: count, dtype: int64

water_source (7 unique):
water_source
River        81550
Open Well    69510
Piped        68993
Borewell     65879
Pond         64113
Tanker       31077
Rainwater    23952
Name: count, dtype: int64

water_treatment (4 unique):
water_treatment
Unt

### Insight
- `state` has 8 unique values — **Assam dominates** with 96,559 records (23.8%), while **Sikkim has the least** with 19,459 (4.8%). This reflects actual population density differences across NE states.
- `district` has 84 unique values with fairly **balanced distribution** (~4,700–5,000 records per district) — no single district overwhelmingly dominates, good for modeling.
- `gender` is perfectly balanced — Female: 202,574 and Male: 202,500. **No gender bias in data collection.**
- `water_source` — **River (81,550) and Open Well (69,510) are the most common** sources, both high-risk for waterborne diseases. Piped water (safest) is only 3rd.
- `water_treatment` — **55% of records use Untreated water (223,691)** — majority of the population is at risk. Clear signal for disease prediction.
- `handwashing_practice` — **Never (195,012) is the most common** practice at 48%. Combined with untreated water, this explains high disease burden.
- `season` — **Monsoon has the highest records (158,992)** followed by Post-Monsoon (98,643). Aligns with known waterborne disease seasonality in Northeast India.
- `disease` — **No_Disease is the majority class (113,645 / 28%)**, Typhoid is the most common disease (55,454), and Leptospirosis is the rarest (28,535). **Class imbalance exists**.

## 3. Setup 

In [3]:
# Display settings
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', '{:.2f}'.format)
plt.rcParams['figure.figsize'] = (12, 6)
plt.rcParams['font.size'] = 12
sns.set_style('whitegrid')

## 4. Target Variable Analysis — Disease Distribution

In [6]:
disease_counts = data['disease'].value_counts()
disease_pct = (
    disease_counts / len(data) * 100).round(2)

disease_data = pd.DataFrame({
    'Count': disease_counts,
    'Percentage': disease_pct
})

print('=== DISEASE DISTRIBUTION ===')
print(disease_data)
print(f'\n Class imbalance ratio (max/min): {disease_counts.max() / disease_counts.min():.2f}x')

=== DISEASE DISTRIBUTION ===
                Count  Percentage
disease                          
No_Disease     113645       28.06
Typhoid         55454       13.69
Dysentery       49956       12.33
Giardiasis      46087       11.38
Cholera         42565       10.51
Hepatitis_E     34769        8.58
Hepatitis_A     34063        8.41
Leptospirosis   28535        7.04

 Class imbalance ratio (max/min): 3.98x


### Insight
- `disease` has 8 unique classes — **No_Disease dominates** with 113,645 records (28.1%), while **Leptospirosis has the least** with 28,535 records (7.0%).
- `disease` — **Typhoid is the most common disease** with 55,454 records (13.7%), followed by Dysentery (49,956) and Giardiasis (46,087).
- `disease` — **Waterborne diseases account for a significant share of cases**, with Typhoid, Dysentery, Giardiasis, and Cholera contributing nearly 48% of all records.
- `disease` — The maximum-to-minimum class ratio is **3.98x**, indicating **moderate class imbalance** but no severely underrepresented disease class.
- `disease` — Disease classes are reasonably well distributed, making the dataset suitable for **multiclass classification modeling**.